src/app/containers/ChartbeatAnalytics/canonical/index.jsx

```diff
@@ -10,21 +10,47 @@ const CanonicalChartbeatBeacon = ({
   chartbeatUID,
   useCanonical,
   chartbeatSource,
-}) => (
-  <Helmet>
-    <script async type="text/javascript">
-      {`
+  hasCookie,
+  title,
+  referrer,
+  hasReferrer,
+}) => {
+  const optionalConfigValues = {
+    ...(hasReferrer && { virtualReferrer: referrer }),
+    ...(hasCookie && { idSync: { bbc_hid: cookie } }),
+  };
+
+  useEffect(() => {
+    if (typeof window !== 'undefined' && window.pSUPERFLY) {
+      window.pSUPERFLY.virtualPage({
```

Comment:
What is this trying to do?

Reply:
this makes sure that that the config is updated when any of the values change

Reply:
https://chartbeat.zendesk.com/hc/en-us/articles/210271287-Handling-virtual-page-changes